### DISCUSSION TITLE
Generating binary BIN file size issues

### DISCUSSION THREAD

**Discussion by www97221** [df8oe/UHSDR#1947] [General]
Thank you for the code file provided by the author. I have a question about how to compress binary files generated by eclipse when using a firmware version that exceeds 600K bytes and only has a firmware size of around 300K bytes

**Comment by arekm** [comment]
Similar question here. Backing up old firmware via USB pendrive method gives me quite big file:

````
% ls -l FW-OVI40.OLD
-rw-r--r--@ 1 arekm  staff  1703936 28 Oct 19:02 FW-OVI40.OLD
# 1.6M
````

While building UHSDR git (via cmdline) firmware file is way smaller:

````
$ ls -l fw-mchf.bin
-rwx------ 1 arekm users 447716 Oct 12 16:21 fw-mchf.bin
# 438K
````

Is FW-OVI40.OLD uncompressed and should be recompressed to be able to upload it again (via usb pendrive) to transceiver somehow
or maybe it contains more data than just firmware?

**Comment by db4ple** [comment]
Hi,this is not a bug, it works as implemented. As we have no safe way of knowing how much space the old firmware was using in the bootloader, the bootloder writes the „whole“ flash memory to the drive. As most build are not taking the full flash, you see a difference. But for the purpose of restoring the previous state, this does not make a difference. Be aware that „whole“ means everything UHSDR would use on a given hardware. This is less than the full memory as UHSDR reserves some space for the storage of configuration data (384k on H7 processors, 64k on other models)Which means your processor is a 2MB H7 model…73DaniloDB4PLEAm 28.10.2024 um 14:21 schrieb Arkadiusz Miśkiewicz ***@***.***>:﻿
Similar question here. Backing up old firmware via USB pendrive method gives me quite big file:
% ls -l FW-OVI40.OLD
-rw-r--r--@ 1 arekm  staff  1703936 28 Oct 19:02 FW-OVI40.OLD
# 1.6M

While building UHSDR git (via cmdline) firmware file is way smaller:
$ ls -l fw-mchf.bin
-rwx------ 1 arekm users 447716 Oct 12 16:21 fw-mchf.bin
# 438K

Is FW-OVI40.OLD uncompressed and should be recompressed to be able to upload it again (via usb pendrive) to transceiver somehow
or maybe it contains more data than just firmware?

**Reply by arekm** [reply]
Thanks.

Additional question:
> UHSDR reserves some space for the storage of configuration data (384k on H7 processors, 64k on other models)

I assume that data is user settings and possibly calibration data? Can these be saved and later restored, too somehow? (in case if someone messes with transceiver software too much). Basically my goal is to be able to fully restore transceiver to the state I have it in now.

**Comment by db4ple** [comment]
Hi,while we are happy for your interest, please understand that this is an issue tracker, not a help forum. We have that, too. Please checkout the Readme, it has all the pointers.TL;DR: Answer is yes, see the manual.73DaniloDB4PLEAm 28.10.2024 um 15:29 schrieb Arkadiusz Miśkiewicz ***@***.***>:﻿
Thanks.
Additional question:

UHSDR reserves some space for the storage of configuration data (384k on H7 processors, 64k on other models)

I assume that data is user settings and possibly calibration data? Can these be saved and later restored, too somehow? (in case if someone messes with transceiver software too much). Basically my goal is to be able to fully restore transceiver to the state I have it in now.

**Reply by arekm** [reply]
Ehm, we are talking in https://github.com/df8oe/UHSDR/discussions not at /issues. Going to dig into manual then.

Edit:

Quick summary to what I found (which may be incorrect) if anyone else ends up finding this discussion.

Backup and restore of settings is described here (my assumption is that this includes calibration data):
https://github.com/df8oe/UHSDR/wiki/Backup-and-Restore

If EEPROM is installed then all settings and calibration is stored there (and not in MCU reserved space). If no EEPROM then MCU reserved space is used.

uhsdr_tool.py backups/restores to what current hardware is using (EEPROM vs reserved space).

**Comment by db4ple** [comment]
Hi,Sorry, my bad!  73DaniloDB4PLEAm 28.10.2024 um 16:04 schrieb Arkadiusz Miśkiewicz ***@***.***>:﻿
Ehm, we are talking in https://github.com/df8oe/UHSDR/discussions not at /issues. Going to dig into manual then.
